google/ads/google_ads/client.py

```diff
@@ -323,17 +331,121 @@ def intercept_unary_unary(self, continuation, client_call_details, request):
 class LoggingInterceptor(grpc.UnaryUnaryClientInterceptor):
     """An interceptor that logs rpc requests and responses."""
 
+    _FULL_REQUEST_LOG_LINE = ('Request\n-------\nMethod: %s\nHost: %s\n'
+                              'Headers: %s\nRequest: %s\n\nResponse\n-------\n'
+                              'Headers: %s\nResponse: %s\n')
+    _FULL_FAULT_LOG_LINE = ('Request\n-------\nMethod: %s\nHost: %s\n'
+                            'Headers: %s\nRequest: %s\n\nResponse\n-------\n'
+                            'Headers: %s\nFault: %s\n')
+    _SUMMARY_LOG_LINE = ('Request made: ClientCustomerId: %s, Host: %s, '
+                         'Method: %s, RequestId: %s, IsFault: %s, '
+                         'FaultMessage: %s')
+
+    def __init__(self, logging_config=None, endpoint=None):
+        """Initializer for the LoggingInterceptor
+
+        Args:
+            logging_config: configuration dictionary for logging
+            endpoint: a str specifying an optional alternative API endpoint.
+        """
+        self.endpoint = endpoint
+        if logging_config:
+            logging.config.dictConfig(logging_config)
+
+    def log_successful_request(self, client_call_details, request, response):
```

Comment:
Much of this could probably be skipped if, for example, the logger is disabled. One check might be to use `Logger.isEnabledFor`.